### PR TITLE
mrsh: init at 2020-01-08

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/mrsh/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromGitHub, meson, ninja, pkgconfig, readline }:
+
+stdenv.mkDerivation rec {
+  pname   = "mrsh";
+  version = "ef21854fc9ce172fb1f7f580b19a89d030d67c65";
```
Comment (filalex77) [inline pkgs/shells/mrsh/default.nix]: Consider keeping a visible `version` as the date of the snapshot...

```suggestion
  version = "2020-01-08";
```

File: pkgs/shells/mrsh/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "emersion";
+    repo = "mrsh";
+    rev = version;
```
Comment (filalex77) [inline pkgs/shells/mrsh/default.nix]: ...and specifying the commit hash only in `fetchFromGitHub`.

```suggestion
    rev = "ef21854fc9ce172fb1f7f580b19a89d030d67c65";
```

File: pkgs/shells/mrsh/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "1iyxmwl61p2x9v9b22416n4lnrlwjqyxybq35x8bcbjxkwypp943";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
+  buildInputs = [ readline ];
+
+  meta = with stdenv.lib; {
+    description = "A minimal POSIX shell";
+    homepage = https://mrsh.sh;
```
Comment (filalex77) [inline pkgs/shells/mrsh/default.nix]: Quoting URLs is preferred now.

```suggestion
    homepage = "https://mrsh.sh";
```

**Review by Br1ght0ne — COMMENTED** [review]
A few stylistic nitpicks, other than that - LGTM

```
[3 built, 1 copied (0.1 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/77307
1 package built:
mrsh
```

**Comment by doronbehar** [comment]
Also the title of the PR should match the one of the commit.

**Review by Br1ght0ne — APPROVED** [review]
LGTM

**Comment by matthiasbeyer** [comment]
@GrahamcOfBorg build mrsh

**Pull request merged**